astropy/coordinates/tests/test_erfa_astrom.py

```diff
@@ -0,0 +1,140 @@
+import numpy as np
+import pytest
+
+
+def test_science_state():
+    import astropy.units as u
+    from astropy.coordinates.erfa_astrom import (
+        erfa_astrom, ErfaAstrom, ErfaAstromInterpolator
+    )
+
+    assert erfa_astrom.get().__class__ is ErfaAstrom
+
+    res = 300 * u.s
+    with erfa_astrom.set(ErfaAstromInterpolator(res)):
+        assert isinstance(erfa_astrom.get(), ErfaAstromInterpolator)
+        erfa_astrom.get().mjd_resolution == res.to_value(u.day)
+
+    # context manager should have switched it back
+    assert erfa_astrom.get().__class__ is ErfaAstrom
+
+    # must be a subclass of BaseErfaAstrom
+    with pytest.raises(TypeError):
+        erfa_astrom.set('foo')
+
+
+def test_warnings():
+    import astropy.units as u
+    from astropy.coordinates.erfa_astrom import (
+        erfa_astrom, ErfaAstromInterpolator
+    )
+    from astropy.utils.exceptions import AstropyWarning
+
+    with pytest.warns(AstropyWarning):
+        with erfa_astrom.set(ErfaAstromInterpolator(9 * u.us)):
+            pass
+
+
+def test_erfa_astrom():
+    # I was having a pretty hard time in coming
+    # up with a unit test only testing the astrom provider
+    # that would not just test its implementation with its implementation
+    # so we test a coordinate conversion using it
+
+    from astropy.coordinates.erfa_astrom import (
+        erfa_astrom, ErfaAstromInterpolator,
+    )
+    from astropy.coordinates import SkyCoord, EarthLocation, AltAz
+    from astropy.time import Time
+    import astropy.units as u
+
+    location = EarthLocation(
+        lon=-17.891105 * u.deg,
+        lat=28.761584 * u.deg,
+        height=2200 * u.m,
+    )
+    obstime = Time('2020-01-01T18:00') + np.linspace(0, 12, 1000) * u.hour
+
+    altaz = AltAz(location=location, obstime=obstime)
+    coord = SkyCoord(ra=83.63308333, dec=22.0145, unit=u.deg)
+
+    # do the reference transformation, no interpolation
+    ref = coord.transform_to(altaz)
+
+    with erfa_astrom.set(ErfaAstromInterpolator(300 * u.s)):
+        interp_300s = coord.transform_to(altaz)
+
+    # make sure they are actually different
+    assert np.any(ref.separation(interp_300s) > u.Quantity(0.01, u.microarcsecond))
```

Comment:
Just `0.01 * u.microarcsecond` is clearer than creating a `Quantity` explicitly (same below).